tests/map/map_copy.cmake

```diff
@@ -0,0 +1,87 @@
+include(cmake_test/cmake_test)
```

Comment:
What you have is fine for now, but CMake uses directories for namespacing. So the pattern `map/copy.cmake`, `list/copy.cmake`, etc. is going to be fairly common. Can we open an issue describing what needs to change so that the path of the test is used for namespacing purposes?

Reply:
I need to look a bit closer at what needs to change exactly, the problem is CTest returned an error because of multiple tests with the same name, we might be able to prepend the test name with the directory path in CMakeTest. I'll open an issue either later today or early tomorrow